rollup/internal/controller/watcher/batch_proposer.go

```diff
@@ -197,16 +208,7 @@ func (p *BatchProposer) proposeBatch() error {
 	var batch encoding.Batch
 	batch.Index = dbParentBatch.Index + 1
 	batch.ParentBatchHash = common.HexToHash(dbParentBatch.Hash)
-	parentBatchEndBlockNumber := daChunks[0].Blocks[0].Header.Number.Uint64() - 1
-	parentBatchCodecVersion := encoding.CodecV0
-	// Genesis batch uses codecv0 encoding, otherwise using bernoulli fork to choose codec version.
-	if dbParentBatch.Index > 0 && p.chainCfg.IsBernoulli(new(big.Int).SetUint64(parentBatchEndBlockNumber)) {
-		parentBatchCodecVersion = encoding.CodecV1
-	}
-	batch.TotalL1MessagePoppedBefore, err = utils.GetTotalL1MessagePoppedBeforeBatch(dbParentBatch.BatchHeader, parentBatchCodecVersion)
-	if err != nil {
-		return err
-	}
+	batch.TotalL1MessagePoppedBefore = firstUnbatchedChunk.TotalL1MessagesPoppedBefore
```

Comment:
Previously we recomputed `TotalL1MessagePoppedBefore`, but actually we can simply use the value from DB?

Reply:
Yes. This info is stored in db, so no need to decode batch encoding here.